test/unit/specs/util/contentfulUtils.spec.js

```diff
@@ -0,0 +1,263 @@
+				path: '/cc/promo-campaign-test',
+				pageType: 'corporate-campaign',
+				pageLayout: {
+					name: 'Promo Campaign Test',
+					contentGroups: [{
+						key: 'promo-campaign-test-cg',
+						name: 'Promo Campaign Test Content Groups',
+						contents: [{
+							key: 'header-area',
+							name: 'Test Campaign Title',
+							bodyCopy: {
+								content: [{
+									content: [{}],
+									data: {},
+									nodeType: 'paragraph',
+								}],
+								data: {},
+								nodeType: 'document',
+							},
+						},
+						{
+							name: 'COVID fund Homepage Promo Images',
+							description: 'Lend a hand from afar and join the global COVID-19 response. Take action',
+							images: expect.any(Array)
+						}]
+					}]
+				}
+			}
+		};
+
+		test('should return page, layout, and content group data', () => {
+			expect(processPageContent(ccPageRaw)).toMatchObject(pageLevelLayoutAndContentGroups);
+		});
+	});
+
+	describe('processPageContentFlattened', () => {
+		test('should return an error object if content is not a page', () => {
+			expect(processPageContentFlat(inactiveUiSetting)).toEqual({ error: 'Non-Page Type Contentful Response' });
+		});
+
+		const pageLevelData = {
+			page: {
+				key: 'promo-campaign-test',
+				path: '/cc/promo-campaign-test',
+				pageType: 'corporate-campaign',
+			}
+		};
+
+		test('should return page level data', () => {
+			expect(processPageContentFlat(ccPageRaw)).toMatchObject(pageLevelData);
+		});
+
+		const pageLevelAndLayoutData = {
+			page: {
+				key: 'promo-campaign-test',
+				path: '/cc/promo-campaign-test',
+				pageType: 'corporate-campaign',
+				pageLayout: {
+					name: 'Promo Campaign Test'
+				},
+				settings: expect.any(Array),
+				contentGroups: expect.any(Object)
+			}
+		};
+
+		test('should return flattened page level + page layout, content groups and content data', () => {
+			expect(processPageContentFlat(ccPageRaw)).toMatchObject(pageLevelAndLayoutData);
+		});
+
+		const pageLevelLayoutAndContentGroups = {
+			page: {
+				key: 'promo-campaign-test',
+				path: '/cc/promo-campaign-test',
+				pageType: 'corporate-campaign',
+				pageLayout: {
+					name: 'Promo Campaign Test',
+				},
+				settings: expect.any(Array),
+				contentGroups: {
+					promo_campaign_test_cg: {
+						key: 'promo-campaign-test-cg',
+						name: 'Promo Campaign Test Content Groups',
+						contents: [{
+							key: 'header-area',
+							name: 'Test Campaign Title',
+							bodyCopy: expect.any(Object)
+						},
+						{
+							name: 'COVID fund Homepage Promo Images',
+							description: 'Lend a hand from afar and join the global COVID-19 response. Take action',
+							images: expect.any(Array)
+						}],
+					}
+				}
+			}
+		};
+
+		test('should return flattened page, layout, and content group data', () => {
+			expect(processPageContentFlat(ccPageRaw)).toMatchObject(pageLevelLayoutAndContentGroups);
+		});
+	});
+
+	describe('processPageContentFlattened with non specific data', () => {
```

Comment:
@emuvente @eddieferrer Kinda feel like I should reduce all the other tests to follow this paradigm...